### PR TITLE
Start filling in ManagePositions page

### PR DESCRIPTION
**Pull request by ptare**
Several fields in the details dropdown aren't filled in yet.

This PR replaces the contract address with the token name. The contract address is still available in the details dropdown.

Issue #595.

### CODE REVIEW

File: sponsor-dapp-v2/src/lib/routes.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const routes = [
   { path: "/Start", component: StartScreen, exact: false },
   { path: "/Steps", component: Steps, exact: false },
   { path: "/ViewPositions", component: ViewPositions, exact: false },
-  { path: "/ManagePositions", component: ManagePositions, exact: false },
+  { path: "/ManagePositions/:tokenAddress", component: ManagePositions, exact: false },
```
Comment (mrice32) [inline sponsor-dapp-v2/src/lib/routes.js]: Just for my information, what does the colon do here?

File: sponsor-dapp-v2/src/views/ManagePositions.js [hunk 2]
```diff
@@ -1,21 +1,142 @@
...
-    const { address } = managePosition.contractStatus;
+  const data = useFinancialContractData(props.match.params.tokenAddress);
+  if (!data.ready) {
+    return <div>Loading data</div>;
+  }
+
+  const numDisplayedDecimals = 4;
+  const format = valInWei => formatWithMaxDecimals(formatWei(valInWei, web3), numDisplayedDecimals, false);
 
+  // This function is only added temporarily to make the diff more readable. Otherwise, github's diff tool isn't able
```
Comment (mrice32) [inline sponsor-dapp-v2/src/views/ManagePositions.js]: Haha, nice trick.

File: sponsor-dapp-v2/src/lib/contracts.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from "react";
+import { drizzleReactHooks } from "drizzle-react";
+
+function useDynamicallyAddedContract(contractAddress, abi) {
+  const { drizzle } = drizzleReactHooks.useDrizzle();
+  const { contract } = drizzleReactHooks.useDrizzleState(drizzleState => ({
+    contract: drizzleState.contracts[contractAddress]
+  }));
+
+  useEffect(() => {
```
Comment (mrice32) [inline sponsor-dapp-v2/src/lib/contracts.js]: Could we limit how often this runs by adding an array of dependencies (maybe adding `contract`, `contractAddress`, and `abi` or something?

File: sponsor-dapp-v2/src/lib/contracts.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from "react";
+import { drizzleReactHooks } from "drizzle-react";
+
+function useDynamicallyAddedContract(contractAddress, abi) {
+  const { drizzle } = drizzleReactHooks.useDrizzle();
+  const { contract } = drizzleReactHooks.useDrizzleState(drizzleState => ({
```
Comment (mrice32) [inline sponsor-dapp-v2/src/lib/contracts.js]: Should this be recomputed whenever contractAddress changes?

Comment (ptare) [reply sponsor-dapp-v2/src/lib/contracts.js]: Done.

File: sponsor-dapp-v2/src/lib/contracts.js (hunk 3 above)
Comment (ptare) [reply sponsor-dapp-v2/src/lib/contracts.js]: Done.

File: sponsor-dapp-v2/src/views/ManagePositions.js [hunk 5]
```diff
@@ -1,21 +1,142 @@
...
+        stateText: "Emergency",
+        stateColor: "red"
+      };
+    case "5":
+      return {
+        stateText: "Settled",
+        stateColor: "blue"
+      };
+    default:
+      throw new Error("oh no");
```
[the remark this answers is not in the document]
Comment (ptare) [reply sponsor-dapp-v2/src/views/ManagePositions.js]: Done.

File: sponsor-dapp-v2/src/lib/routes.js (hunk 1 above)
Comment (ptare) [reply sponsor-dapp-v2/src/lib/routes.js]: It matches whatever is in the URL after `/ManagePositions/` and provides it as a variable called `tokenAddress`.

Comment (mrice32) [reply sponsor-dapp-v2/src/lib/routes.js]: So cool!

**Pull request merged**